docs/data/date-pickers/base-concepts/ControlledComponent.tsx

```diff
@@ -11,7 +11,12 @@ export default function ControlledComponent() {
   return (
     <LocalizationProvider dateAdapter={AdapterDayjs}>
       <DemoContainer components={['DatePicker']}>
-        <DatePicker value={value} onChange={(newValue) => setValue(newValue)} />
+        <DatePicker
+          value={value}
+          onChange={(newValue) => {
+            setValue(newValue);
```

Comment:
Make it clear that there is no return value (for the noobies)

Reply:
Or go the other—minimalism way. 🙈 😆 
```tsx
onChange={setValue}
```
In any case, this only changes one example, albeit, in the the getting started page. Do you think it is enough and leaving others as is doesn't hurt? 🤔

Reply:
> onChange={setValue}

I don't think that we can do this. I would expect more arguments, starting with the `event`.

> Do you think it is enough and leaving others as is doesn't hurt?

I was thinking of one change at a time. It's not a huge difference, but I tend to make this change anytime I came across it.

Reply:
> I don't think that we can do this. I would expect more arguments, starting with the `event`, and later the reason why the onChange event triggers.

Fair point. We are also providing the context as the second argument, possibly containing validation errors.

> It's not a huge difference, but I tend to make this change anytime I came across it.

I've gone through all v6 documentation for pickers and couldn't find a single case where we are using your suggested format.
I do understand the reason why you'd prefer it, but I believe that @flaviendelangle has chosen the shorter format to be have shorter demo previews.
IMHO, if we change the format, I'd vote for changing every usage of it to have consistency.

Reply:
In the pickers maybe, but I I think that in the overall scope, we default to be explicit about the returned values, e.g. https://mui.com/material-ui/react-text-field/#uncontrolled-vs-controlled. I'm reverting here as it's outside of the scope of the bug fix.